zerver/views/registration.py

```diff
@@ -325,6 +325,7 @@ def redirect_to_email_login_url(email: str) -> HttpResponseRedirect:
     redirect_url = login_url + '?already_registered=' + email
     return HttpResponseRedirect(redirect_url)
 
+@validate_recaptcha
```

Comment:
Hmm. I know making this a decorator is kind of like how we do some other validation checks on other views, but it bothers me here because it feels pretty far separated from where we actually decide whether this form submission is OK.

I think where I'd rather put this is inside the Django form validation, so in `RealmCreationForm` in `zerver/forms.py`. So basically the same logic as in the core of this `validate_recaptcha` function, except it raises a `ValidationError` on failure. Can you switch it to that?

Reply:
Sure. I would be inclined to make a function and use it in the form so that it can be used in other forms as well.

Reply:
Sure. We'll definitely want that when we start using it in other forms.